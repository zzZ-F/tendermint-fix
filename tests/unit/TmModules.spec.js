import { shallowMount } from "@vue/test-utils"
import htmlBeautify from "html-beautify"
import { TmModules } from "@/index"
import { TmModule } from "@/index"

describe("TmModules.vue", () => {
  let wrapper
  beforeEach(async () => {
    wrapper = shallowMount(TmModules, {
      slots: { default: "<div>slot</div>" }
    })
  })

  it("has the expected html structure", () => {
    expect(htmlBeautify(wrapper.html())).toMatchSnapshot()
  })
})
describe("TmModule.vue", () => {
  let wrapper
  beforeEach(async () => {
    wrapper = shallowMount(TmModule, {
      slots: {
        title: "<div>title</div>",
        menu: "<div>menu</div>",
        default: "<div>default</div>",
        footer: "<div>footer</div>"
      }
    })
  })

  it("has the expected html structure", () => {
    expect(htmlBeautify(wrapper.html())).toMatchSnapshot()
  })
})
