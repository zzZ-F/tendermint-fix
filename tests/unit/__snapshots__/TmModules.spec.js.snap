// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmModule.vue has the expected html structure 1`] = `
<div class="tm-module">
  <div class="tm-module-container">
    <header class="tm-module-header">
      <div class="tm-module-title">
        <div>
          title
        </div>
      </div>
      <div class="tm-module-menu">
        <div>
          menu
        </div>
      </div>
    </header>
    <main class="tm-module-main">
      <div>
        default
      </div>
    </main>
    <footer class="tm-module-footer">
      <div>
        footer
      </div>
    </footer>
  </div>
</div>
`;

exports[`TmModules.vue has the expected html structure 1`] = `
<div class="tm-modules">
  <div>
    slot
  </div>
</div>
`;
